scala/README.md

```diff
@@ -382,21 +441,41 @@ It is definitely recommended to read the full Twitter's "[Effective Scala]" guid
 
     1. Instead of `c exists (_ == 0)`, use `c contains 0`.
 
-0. Do not import `scala.collection.mutable._` or even a single mutable collection directly. Instead, import the `mutable` package and use it explicitly as a namespace prefix to denote mutability:
+    This is not only a matter of style: for some collections, the recommendations above can be algorithmically more efficient than the naive approach.
+
+0. Do not import `scala.collection.mutable._` or even any single mutable collection directly. Instead, import the `mutable` package itself and use it explicitly as a namespace prefix to denote mutability, which also avoids name conflicts if using both mutable and immutable structures in the same scope:
 
     ```scala
-    // Bad, too subtle and risky for the inattentive reader.
     import scala.collection.mutable.Set
-    val set = Set(1, 2, 3)
+    val bad = Set(1, 2, 3) // Too subtle and risky for the inattentive reader
 
-    // Better
     import scala.collection.mutable
-    val set = mutable.Set(1, 2, 3)
+    val good = mutable.Set(1, 2, 3)
     ```
 
-0. No "stringly" typed code. Use `Enumeration` or `sealed` types and `case object`s. In many cases, `Enumeration` and `sealed` types are interchangeable, but they do not fully overlap. `Enumeration`, for instance, does not check for exhaustive matching while `sealed` types do not, well, enumerate.
+0. No "stringly" typed code. Use `Enumeration` or `sealed` types with `case` objects. `Enumeration` and `sealed` types have similar purpose and usage, but they do not fully overlap. `Enumeration`, for instance, does not check for exhaustive matching while `sealed` types do not, well, enumerate.
+
+    ```scala
+    object Season extends Enumeration {
+      type Season = Value
+      val Spring, Summer, Autumn, Winter = Value
+    }
+
+    sealed trait Gender
+    case object Male extends Gender
+    case object Female extends Gender
+    ```
 
-0. If an API is not 100% "typesafe" (for instance, all parameters are of type `Int` or `String`), always use named parameters to disambiguate:
+0. When creating new `Enumeration`s, always define a type alias (as above). Never use `MyEnum.Value` to refer to the type of enumeration values:
+
+    ```scala
+    def bad(a: Season.Value) = ???
+
+    import Season.Season
+    def good(a: Season) = ???
+    ```
+
+0. If an API does not differentiate its function parameters with proper typing (for instance, multiple parameters are `Int` or `String`), always use named parameters to disambiguate:
```

Comment:
though the way you phrase it, it makes it sound like the person who wrote the function is being improper, but i feel like it should be ok to write a function that takes multiple values of the same type. 

maybe rephrase to "if a function takes multiple arguments of the same type, use named parameters to ensure you do not mistakenly pass a value to 'lat' that was actually your 'lng'"